File: go/problems/100_same_tree_test.go
package main

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestIsSameTree(t *testing.T) {
	tests := []struct {
		p    *TreeNode
		q    *TreeNode
		want bool
	}{
		{
			p: &TreeNode{
				Val: 1,
				Left: &TreeNode{
					Val: 2,
				},
				Right: &TreeNode{
					Val: 3,
				},
			},
			q: &TreeNode{
				Val: 1,
				Left: &TreeNode{
					Val: 2,
				},
				Right: &TreeNode{
					Val: 3,
				},
			},
			want: true,
		},
		{
			p: &TreeNode{
				Val: 1,
				Left: &TreeNode{
					Val: 2,
				},
			},
			q: &TreeNode{
				Val: 1,
				Right: &TreeNode{
					Val: 2,
				},
			},
			want: false,
		},
		{
			p: &TreeNode{
				Val: 1,
				Left: &TreeNode{
					Val: 2,
				},
				Right: &TreeNode{
					Val: 1,
				},
			},
			q: &TreeNode{
				Val: 1,
				Left: &TreeNode{
					Val: 1,
				},
				Right: &TreeNode{
					Val: 2,
				},
			},
			want: false,
		},
	}

	for _, tt := range tests {
		t.Run(fmt.Sprintf("given trees %v and %v", tt.p, tt.p), func(t *testing.T) {
			got := isSameTree(tt.p, tt.q)
			assert.Equal(t, tt.want, got, "got %t want %t", got, tt.want)
		})
	}
}

File: ts/problems/100-same-tree.test.ts
import { describe, expect, test } from "vitest";
import { isSameTree } from "./100-same-tree";

describe("isSameTree", () => {
  test.each([
    {
      p: {
        val: 1,
        left: {
          val: 2,
          left: null,
          right: null,
        },
        right: {
          val: 3,
          left: null,
          right: null,
        },
      },
      q: {
        val: 1,
        left: {
          val: 2,
          left: null,
          right: null,
        },
        right: {
          val: 3,
          left: null,
          right: null,
        },
      },
      want: true,
    },
  ])("given trees $p and $q", ({ p, q, want }) => {
    expect(isSameTree(p, q)).toBe(want);
  });
});
